packages/amplify-graphql-conversation-transformer/src/resolvers/assistant-mutation-resolver.ts

```diff
@@ -11,7 +12,15 @@ import { ConversationDirectiveConfiguration } from '../grapqhl-conversation-tran
  * @returns {MappingTemplateProvider} An object containing request and response MappingTemplateProviders.
  */
 export const assistantMutationResolver = (config: ConversationDirectiveConfiguration): MappingTemplateProvider => {
-  const resolver = fs.readFileSync(path.join(__dirname, 'assistant-mutation-resolver-fn.template.js'), 'utf8');
+  let resolver = fs.readFileSync(path.join(__dirname, 'assistant-mutation-resolver-fn.template.js'), 'utf8');
+  const fieldName = toUpper(config.field.name.value);
+  const substitutions = {
+    CONVERSATION_MESSAGE_TYPE_NAME: `ConversationMessage${fieldName}`,
+  };
+  Object.entries(substitutions).forEach(([key, value]) => {
+    const replaced = resolver.replace(new RegExp(`\\[\\[${key}\\]\\]`, 'g'), value);
```

Comment:
OK, I'm calling it: we need a central method for template substitution. Please refactor.

Reply:
It's in progress in the `ai.conversation-consolidate-resolver-generation` branch.

I needed prioritize these changes (this and the other PRs from this/last week) and am allergic to mixing functional and (non-trivial) non-functional changes in PRs. I also wanted to make sure whatever abstraction I use here will cover these changes. Thanks for your patience, I know I've tested it with these last few PRs 😅